### PR TITLE
Delete NCP created resources in cleanup 

### PR DESCRIPTION
**Pull request by yuntanghsu**
In this commit, we delete NCP created resource (share/sharedResource/certificate/lbAppProfile/lbPersistenProfile) in the cleanup process.

### CODE REVIEW

File: pkg/nsx/services/vpc/vpc.go [hunk 1]
```diff
@@ -259,6 +259,196 @@ func (s *VPCService) DeleteVPC(path string) error {
...
+	store := &ResourceStore{ResourceStore: common.ResourceStore{
+		Indexer:     cache.NewIndexer(keyFunc, cache.Indexers{}),
+		BindingType: model.TlsCertificateBindingType(),
+	}}
+	query := fmt.Sprintf("%s:%s", common.ResourceType, common.ResourceType)
+	query = s.addClusterTag(query)
+	count, searcherr := s.SearchResource(common.ResourceTypeTlsCertificate, query, store, nil)
+	if searcherr != nil {
+		log.Error(searcherr, "failed to query certificate", "query", query)
+	} else {
```
Comment (TaoZou1) [inline pkg/nsx/services/vpc/vpc.go:309]: certificate -> profile

File: pkg/nsx/services/vpc/vpc.go [hunk 2]
```diff
@@ -259,6 +259,196 @@ func (s *VPCService) DeleteVPC(path string) error {
...
+	tagScopeClusterKey := strings.Replace(common.TagScopeNCPCluster, "/", "\\/", -1)
+	tagScopeClusterValue := strings.Replace(s.NSXClient.NsxConfig.Cluster, ":", "\\:", -1)
+	tagParam := fmt.Sprintf("tags.scope:%s AND tags.tag:%s", tagScopeClusterKey, tagScopeClusterValue)
+	return query + " AND " + tagParam
+}
+
+func (s *VPCService) ListCert() []model.TlsCertificate {
+	store := &ResourceStore{ResourceStore: common.ResourceStore{
+		Indexer:     cache.NewIndexer(keyFunc, cache.Indexers{}),
+		BindingType: model.TlsCertificateBindingType(),
```
Comment (TaoZou1) [inline pkg/nsx/services/vpc/vpc.go:272]: does it need store? just query and handling the result seems enough.

Comment (TaoZou1) [reply pkg/nsx/services/vpc/vpc.go:272]: using store could reuse the logic. looks fine.

File: pkg/nsx/services/vpc/vpc.go [hunk 3]
```diff
@@ -259,6 +259,196 @@ func (s *VPCService) DeleteVPC(path string) error {
...
+
+func (s *VPCService) ListProfile() []model.TlsCertificate {
+	store := &ResourceStore{ResourceStore: common.ResourceStore{
+		Indexer:     cache.NewIndexer(keyFunc, cache.Indexers{}),
+		BindingType: model.TlsCertificateBindingType(),
+	}}
+	query := fmt.Sprintf("%s:%s", common.ResourceType, common.ResourceType)
+	query = s.addClusterTag(query)
+	count, searcherr := s.SearchResource(common.ResourceTypeTlsCertificate, query, store, nil)
+	if searcherr != nil {
```
[the remark this answers is not in the document]
Comment (yuntanghsu) [reply pkg/nsx/services/vpc/vpc.go]: Thanks for reviewing.
This func is unused. I forget to delete it.
Removed it. Thanks.

**Review by dantingl — DISMISSED** [review]
Yun-Tang, please rebase your patch

**Review by dantingl — DISMISSED** [review]

**Review by salv-orlando — APPROVED** [review]
Approving for triggering workflow run

**Review by salv-orlando — APPROVED** [review]

**Comment by salv-orlando** [comment]
> Approving for triggering workflow run

@yuntanghsu you need approval from a maintainer, unfortunately

**Review by dantingl — APPROVED** [review]

**Review by TaoZou1 — APPROVED** [review]

**Pull request merged**